### PR TITLE
Add a "Show Password" Checkbox to the login and register pages

### PR DESCRIPTION
**Pull request by bstutes**
Show password input (checkbox) #1984

This pull request updates/supersedes my pull request of June 9, 2017.

Details of changes follow.

NOTE: All "-js-pw-m-*" CSS selectors referenced are
for use by the password reveal JS event handler
code.

.../tests/unit:
test_csp.py:
Updated script-src assert statement spec to include "ajax.googleapis.com".

.../warehouse/templates/account:
login.html, register.html:
Added "-js-pw-m-fld" CSS selector to form.password
class_ string for PW reveal/hide toggle.
Added label and checkbox elements for triggering
password reveal function. Checkbox element
identified by "-js-pw-m-cbx" CSS selector for
cleanup form on submit.
Added "-js-pw-m-sbt" selector to form submit
button used by JS event handler to trigger
form cleanup on submit.
Added "extra_js" block to insert "jQuery" and
"pwmask" script tags for password reveal JS code.
In register.html, "for=" attribute for the email input
was incorrect. Changed from "full_name" to "email".

.../warehouse/static/sass/blocks/_form-group.scss:
Added &__show-pw selector for positioning the show_pw checkbox and text.

.../warehouse/static/js:
pwmask/pw-mask-toggle.js
JS containing event handlers for password
reveal/hide toggle and submit cleanup.

.../warehouse:
csp.py:
Added "ajax.googleapis.com" script-src policy
statement to allow loading jquery.min.js in
login.html and register.html.

.../Gulpfile.babel.js:
Added path.join statement for "pw-mask-toggle.js"
to files list in gulp.task("dist:js", ...) function.
Added "else if (relPath..." statement for naming
minified JS code for password reveal toggling.

### CODE REVIEW

**Comment by di** [comment]
Hi @bstutes, we went with a somewhat simpler approach in #2661 so I'm going to close this, but thanks very much for giving this a shot!
